compiler/rustc_typeck/src/collect.rs

```diff
@@ -2876,14 +2876,9 @@ fn codegen_fn_attrs(tcx: TyCtxt<'_>, id: DefId) -> CodegenFnAttrs {
                     InlineAttr::None
                 } else if list_contains_name(&items[..], sym::always) {
                     if tcx.sess.instrument_coverage() {
```

Comment:
Looking at this again, I'd prefer if this fix isn't in typeck. From a compiler architecture standpoint, that means coverage flags can change semantics of the program.

Looking at all the references to `InlineAttr` there are a handful of places in the backend where it gets used. Maybe a helper method would be good here. I'd also accept a FIXME to address this later.

Reply:
Actually a FIXME might be preferred, since this is much simpler. I'm not 100% sure what the right answer is here.

Reply:
I think I'll add the FIXME. I searched a variety of locations, but this one was the most central that I could find.

Reply:
added #83429